instructor/process_response.py

```diff
@@ -443,117 +514,114 @@ def handle_response_model(
 
 The output must be a valid JSON object that `{response_model.__name__}.model_validate_json()` can successfully parse.
 """
-            messages = new_kwargs.pop("messages", [])
-            chat_history = []
-            for message in messages:
-                # format in Cohere's ChatMessage format
-                chat_history.append(
-                    {
-                        "role": message["role"],
-                        "message": message["content"],
-                    }
-                )
-            new_kwargs["message"] = instruction
-            new_kwargs["chat_history"] = chat_history
-
-            if "strict" in new_kwargs:
-                del new_kwargs["strict"]
-        elif mode == Mode.GEMINI_JSON:
-            assert (
-                "model" not in new_kwargs
-            ), "Gemini `model` must be set while patching the client, not passed as a parameter to the create method"
-
-            from .utils import update_gemini_kwargs
-
-            message = dedent(
-                f"""
-                As a genius expert, your task is to understand the content and provide
-                the parsed objects in json that match the following json_schema:\n
-
-                {json.dumps(response_model.model_json_schema(), indent=2, ensure_ascii=False)}
-
-                Make sure to return an instance of the JSON, not the schema itself
-                """
-            )
-            # check that the first message is a system message
-            # if it is not, add a system message to the beginning
-            if new_kwargs["messages"][0]["role"] != "system":
-                new_kwargs["messages"].insert(
-                    0,
-                    {
-                        "role": "system",
-                        "content": message,
-                    },
-                )
-            # if it is, system append the schema to the end
-            else:
-                new_kwargs["messages"][0]["content"] += f"\n\n{message}"
-
-            # default to json response type
-            new_kwargs["generation_config"] = new_kwargs.get(
-                "generation_config", {}
-            ) | {"response_mime_type": "application/json"}
-
-            new_kwargs = update_gemini_kwargs(new_kwargs)
-
-        elif mode == Mode.GEMINI_TOOLS:
-            assert (
-                "model" not in new_kwargs
-            ), "Gemini `model` must be set while patching the client, not passed as a parameter to the create method"
-            from .utils import update_gemini_kwargs
-
-            new_kwargs["tools"] = [response_model.gemini_schema]
-            new_kwargs["tool_config"] = {
-                "function_calling_config": {
-                    "mode": "ANY",
-                    "allowed_function_names": [response_model.__name__],
-                },
-            }
+    new_kwargs["chat_history"] = [
```

Comment:
Reverted back to using normal JSON mode - wasn't able to get tool calling working with Cohere, will fix in a separate PR down the line